lib/puppet_x/puppetlabs/strings/yard/code_objects/method_object.rb

```diff
@@ -0,0 +1,34 @@
+class YARD::CodeObjects::MethodObject
```

Comment:
As opposed to patching Yard's `MethodObject` we could just create `Puppet4xFunctionObject` and `Puppet3xFunctionObject`s which inherit from it. There is some logic identifying puppet and ruby methods (`:method`) right now which would need to be modified as a result.